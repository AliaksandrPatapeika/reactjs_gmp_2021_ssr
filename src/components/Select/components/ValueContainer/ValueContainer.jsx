import PropTypes from 'prop-types';
import React from 'react';
import {components} from 'react-select';

const allOption = {
  label: 'Select all',
  value: '*'
};

const ValueContainer = ({children, ...props}) => {
  const {getValue} = props;
  const currentValues = getValue();

  let toBeRendered = children;

  if (currentValues.some((val) => val.value === allOption.value)) {
    toBeRendered = [[children[0][0]], children[1]];
  }

  return (
    <components.ValueContainer {...props}>
      {toBeRendered}
    </components.ValueContainer>
  );
};

ValueContainer.propTypes = {
  getValue: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired
};

export default ValueContainer;
