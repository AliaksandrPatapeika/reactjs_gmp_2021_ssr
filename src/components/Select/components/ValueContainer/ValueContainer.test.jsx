import {noop} from 'lodash';
import React from 'react';
import renderer from 'react-test-renderer';

import ValueContainer from './ValueContainer';

describe('ValueContainer', () => {
  test('renders correctly', () => {
    const valueContainer = renderer
      .create(
        <ValueContainer getStyles={noop} cx={noop} getValue={() => [{value: '*'}]}>
          {[[[<span key="1">children 0</span>]], [<span key="2">children 1</span>]]}
        </ValueContainer>
      )
      .toJSON();

    expect(valueContainer).toMatchSnapshot();
  });
});
