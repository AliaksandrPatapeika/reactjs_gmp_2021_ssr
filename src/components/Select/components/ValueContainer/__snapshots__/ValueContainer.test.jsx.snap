// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValueContainer renders correctly 1`] = `
<div
  className="css-pilx6v-ValueContainer"
>
  <span>
    children 0
  </span>
  <span>
    children 1
  </span>
</div>
`;
